### PR TITLE
Improved Dashboard Error Feedback

### LINKED ISSUE #6285: Dashboard (or insight) fails to load
## Bug description

I quite frequently have problems where either an Insight or Dashboard does not load. In this case, the dashboard does not load.

## Expected behavior

The Dashboard or Insight should load.

If the error is caused by a network issue of some sort (but, as mentioned, this happens a lot) then a different message in the center of the screen may be better.

## How to reproduce

The following video shows the Dashboard as empty. However, it should not be empty.

https://user-images.githubusercontent.com/328367/136211017-d9c85863-f8f9-4fda-a727-0ad2054b0c9b.mp4

## Environment

- [x] PostHog Cloud
- [ ] self-hosted PostHog (ClickHouse-based), version/commit: _please provide_
- [ ] self-hosted PostHog (Postgres-based, legacy), version/commit: _please provide_

### PR DESCRIPTION
**Pull request by pauldambra**
## Changes

Closes #6285 

This introduces three changes

1) makes the loading overlay for insights less subtle so loading is easier to see on slow connections
2) adds an error state for individual insights on a dashboard to distinguish from their empty state
3) adds an error state for an entire dashboard to distinguish from its empty state

### before loading change

before

### after loading change

after

#### before

Screenshot 2021-11-30 at 12 14 46

#### after

Screenshot 2021-11-30 at 12 06 12

#### before

Screenshot 2021-11-30 at 12 16 09

#### after

Screenshot 2021-11-30 at 11 58 55

## How did you test this code?

with some unit tests and manual testing

in order to test manually the easiest route is to load a dashboard then run `redis-cli KEYS "*cache*" | xargs redis-cli DEL` so the cache for the items is removed and then insert a throw in either the `dashboardLogic > loaders > allItems` to `dashboardLogic > refreshAllDashboardItems > fetchItemFunctions` depending on if you want to error on all items or individual items respectively then either refresh the page or click to refresh the dashboard

### CODE REVIEW

File: frontend/src/styles/global.scss [hunk 1]
```diff
@@ -779,3 +779,27 @@ body {
...
+    justify-content: center;
+
+    &.over-table {
+        display: block;
+        background: rgba(0, 0, 0, 0.1);
+        td {
+            display: block;
+        }
+    }
+}
```
Comment (Twixes) [inline frontend/src/styles/global.scss:805]: Why the move from `style.scss`?

File: frontend/src/scenes/dashboard/DashboardItem.tsx [hunk 2]
```diff
@@ -246,8 +248,8 @@ export function DashboardItem({
         }
 
         // Insight agnostic empty states
-        if (showErrorMessage) {
-            return <InsightErrorState />
+        if (showErrorMessage || receivedErrorFromAPI) {
+            return <InsightErrorState excludeDetail={true} />
```
Comment (Twixes) [inline frontend/src/scenes/dashboard/DashboardItem.tsx:252]: `InsightErrorState`'s gray background makes sense in the larger size (`excludeDetail={false}`) but here it looks kind of sloppy:
144045452-e10f5c2f-fde0-42b5-a589-27f35e86ecb3 (1)
I'm going to rely on @clarkus's skills to suggest the better way though.

File: frontend/src/styles/global.scss [hunk 3]
```diff
@@ -779,3 +779,27 @@ body {
...
+.loading-overlay {
+    width: 100%;
+    height: 100%;
+    position: absolute;
+    top: 0;
+    left: 0;
+    background: $bg_charcoal;
+    text-align: center;
+    min-height: 6rem;
+    opacity: 0.7;
```
Comment (Twixes) [inline frontend/src/styles/global.scss:792]:
```suggestion
    opacity: 0.5;
```
Found the level of darkness a bit out of style for us, 0.5 would be more typical. Connected to this is that the spinner should always have `opacity: 1` – it looks a bit ghostly when it has the same transparency as the overlay's background.

**Review by Twixes — CHANGES_REQUESTED** [review]
Big improvement. Code looks good, just some notes on the UI

File: frontend/src/scenes/dashboard/DashboardItem.tsx (hunk 2 above)
Comment (clarkus) [reply frontend/src/scenes/dashboard/DashboardItem.tsx:252]: These errors states do look a little cramped. If we retain the gray background, we should pad the container a bit more. If it's feasible to drop the gray background and reuse colors from the dashboard item color styles, then we wouldn't need to change padding. Put together a quick example:
Frame 9192

**Review by clarkus — APPROVED** [review]
Solid improvements! Left some examples to guide feedback, but I don't see anything blocking from a UX point of view. Nice job, @pauldambra

File: frontend/src/styles/global.scss (hunk 1 above)
Comment (pauldambra) [reply frontend/src/styles/global.scss:805]: The file is listed as deprecated so I thought better not to edit

**Comment by pauldambra** [comment]
# spinner with just background opaquified



# error state matching the container

error-states

**Comment by mariusandra** [comment]
Not a review, but a comment that it would be awesome to (somehow? eventually?) see all these different states, with all the available colors somewhere on a storybook. I started something like this in #7017, but it got stalled.

[Here's the storybook page](https://61260360a8e859003a7c138b-vlsmeipqqt.chromatic.com/?path=/story/posthog-scenes-dashboard--all-possible-insight-types) as it is in that PR. The next steps is to add all the different types of charts with some real-ish data by copying a few API responses  into json files. The step after that is to add other stories where all the charts show a specific state (loading, error, timeout, etc).

I'd be happy if someone could pick that up. 🤞

File: frontend/src/scenes/insights/Insight.scss [hunk 4]
```diff
@@ -314,6 +314,46 @@ $funnel_canvas_background: #fafafa;
...
+
+                    h2 {
+                        color: $text_light;
+                    }
+                }
+            }
+        }
+    }
+
+    &:not(.purple) {
```
Comment (Twixes) [inline frontend/src/scenes/insights/Insight.scss:340]: Hm, not sure about giving _just_ purple special handling here - maybe all non-white should have this styling?

**Review by Twixes — APPROVED** [review]
Looks great. One styling comment

File: frontend/src/scenes/insights/Insight.scss (hunk 4 above)
Comment (pauldambra) [reply frontend/src/scenes/insights/Insight.scss:340]: I might pick up the storybook as described above then. Can check which colours do or don't work with white text then

**Pull request merged**